React components

margotTenenbaum opened the pull request:
Hi @GrumpyCat0709 and @Nick-Zabalza,

This branch builds out all of my needed react components and passes the necessary data to each of them. There is some basic css applied, but the styling is not complete and should not be considered a part of this pull request. 
I used Enzyme with Jest to create basic unit testing for my React components. This ended up being considerably more complicated than anticipated, and due to time-constraints I decided to stop at basic unit testing. I have added a ticket to my trello to add more robust integration tests in the near future.
README is also updated to reflect new functionality.

Thank you!

Nick-Zabalza commented:
@margotTenenbaum 

Hey Elizabeth,

Ok, so here is the deal: I'm exhausted.

I did the code review but I wouldn't say it was my best effort.

Case in point: I forogt to run npm install first thing.

I forgot to run npm install because I was doing the low mental effort thing of following your ReadMe and there is no first step saying "run npm install". I would add that.

I also noticed at least two other issues. Both in your reactTests.test.js file.

1. On line 7, you "import { shallow, mount, render } from 'enzyme';".  On the line above, you already import Enzyme and so this line could be const { shallow, mount, render } = Enzyme; This may or not be a stylistic preference. Seems to me having to search through all the node modules folder for a folder called 'enzyme' takes more effort than directly using the varaible already defined. But i could be wrong about that since i do not know the Big O of the require keyword. Up to you if you want to change this.
2. There seemed to be very little tests. Though I was too tired to determine this myself. However, I believe Jest has a code coverage tool. If it says you are still covering at least 70% of your code, then I guess this is fine. If it says otherwise, more tests are needed before you merge.

Good work!

margotTenenbaum commented:
@Nick-Zabalza 
Thank you! I made your suggested changes in the README and switched to const instead of import (this made the tests run quite a bit faster, so great catch!)
I also added unit tests to ensure 100% coverage of the 4 main react components. The index.jsx coverage is still lower than 70%, but I will address this in integration tests in the near future.
